stats/cloud/api_test.go

```diff
@@ -202,3 +218,30 @@ func TestRetrySuccessOnSecond(t *testing.T) {
 	assert.NotNil(t, resp)
 	assert.Nil(t, err)
 }
+
+func TestIdempotencyKey(t *testing.T) {
+	const idempotencyKey = "xxx"
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotK6IdempotencyKey := r.Header.Get(k6IdempotencyKeyHeader)
+		switch r.Method {
+		case http.MethodPost:
+			assert.NotEmpty(t, gotK6IdempotencyKey)
+			assert.Equal(t, idempotencyKey, gotK6IdempotencyKey)
+		default:
+			assert.Empty(t, gotK6IdempotencyKey)
+		}
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	client := NewClient("token", server.URL, "1.0")
+	client.retryInterval = 1 * time.Millisecond
+	req, err := client.NewRequest(http.MethodPost, server.URL, nil)
+	assert.NoError(t, err)
+	req.Header.Set(k6IdempotencyKeyHeader, idempotencyKey)
```

Comment:
Not sure what's tested here besides the Go HTTP library :wink:

Reply:
It tests that user can supply its own Idempotency-Key

Reply:
Fair enough, though I'm not sure we have a valid use case for that